elements/lisk-codec/src/string.ts

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright © 2020 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+import { writeVarInt } from './varint';
+
+interface SchemaProperty {
+	readonly dataType: string;
+}
+
+export const writeString = (value: string, _schema: SchemaProperty): Buffer =>
+	Buffer.concat([
+		Buffer.from('0a', 'hex'),
+		writeVarInt(value.length, { dataType: 'uint32' }),
+		Buffer.from(value, 'utf8'),
+	]);
+
+export const readString = (buffer: Buffer, _schema: SchemaProperty): string =>
+	buffer.toString('utf8', 2, buffer.length);
```

Comment:
so for this offset is `1` i think, if we dont include the field number

Reply:
so if I use offset to 1 then `Checkout Lisk SDK!` -> `12436865636b6f7574204c69736b2053444b21` which is missing `0a`prefix as compared to protocol-spec.

Reply:
I think protocol spec is including the field number because if you read the whole schema, thats what should include, but for this implementation specifically, the field number should be ignored or skipped

Reply:
Okay, so the key's significance is from outside but not within the string writer. So in that case either I change the protocol spec or simply ignore the key part coming from the protocol-spec output

Reply:
You can ignore the key part of concatenate in the test like simulating the whole encoder.  If you check the protocol-spec it's a complete message that's why the difference. So as you said you could remove the key or add it. Both work :)

Reply:
let's keep the protocol-spec as it is, so that we can test the most outer `encode` and `decode` function using the exact protocol-spec defined

Reply:
Yeah I just pushed the changes and just ignoring the key part from protocol-spec